editor/src/components/inspector/sections/layout-section/self-layout-subsection/frame-updating-layout-section.tsx

```diff
@@ -0,0 +1,271 @@
+  )
+
+  const updateFrame = React.useCallback(
+    (edgePosition: EdgePosition, movement: CanvasVector) => {
+      if (edgePosition === EdgePositionTop || edgePosition === EdgePositionLeft) {
+        executeFirstApplicableStrategy(
+          dispatch,
+          metadataRef.current,
+          selectedViewsRef.current,
+          elementPathTreeRef.current,
+          allElementPropsRef.current,
+          [moveInspectorStrategy(projectContentsRef.current, movement)],
+        )
+      } else {
+        executeFirstApplicableStrategy(
+          dispatch,
+          metadataRef.current,
+          selectedViewsRef.current,
+          elementPathTreeRef.current,
+          allElementPropsRef.current,
+          [
+            resizeInspectorStrategy(
+              projectContentsRef.current,
+              originalGlobalFrame,
+              edgePosition,
+              movement,
+            ),
+          ],
+        )
+      }
+    },
+    [
+      allElementPropsRef,
+      dispatch,
+      elementPathTreeRef,
+      metadataRef,
+      originalGlobalFrame,
+      projectContentsRef,
+      selectedViewsRef,
+    ],
+  )
+
+  return (
+    <>
+      <UIGridRow
+        padded={false}
+        variant='<--1fr--><--1fr-->'
+        style={{ minHeight: undefined, gap: 4 }}
+      >
+        <FrameUpdatingLayoutControl
+          property='left'
+          label='L'
+          updateFrame={updateFrame}
+          currentValue={originalLocalFrame.x}
+          enabled={singleItemSelected}
+        />
+        <FrameUpdatingLayoutControl
+          property='top'
+          label='T'
+          updateFrame={updateFrame}
+          currentValue={originalLocalFrame.y}
+          enabled={singleItemSelected}
+        />
+      </UIGridRow>
+      <UIGridRow
+        padded={false}
+        variant='<--1fr--><--1fr-->'
+        style={{ minHeight: undefined, gap: 4 }}
+      >
+        <FrameUpdatingLayoutControl
+          property='width'
+          label='W'
+          updateFrame={updateFrame}
+          currentValue={originalLocalFrame.width}
+          enabled={singleItemSelected}
+        />
+        <FrameUpdatingLayoutControl
+          property='height'
+          label='H'
+          updateFrame={updateFrame}
+          currentValue={originalLocalFrame.height}
+          enabled={singleItemSelected}
+        />
+      </UIGridRow>
+    </>
+  )
+})
+FrameUpdatingLayoutSection.displayName = 'FrameUpdatingLayoutSection'
+
+interface LayoutPinPropertyControlProps {
+  label: string
+  property: TLWH
+  currentValue: number
+  updateFrame: (edgePosition: EdgePosition, movement: CanvasVector) => void
+  enabled: boolean
+}
+
+const FrameUpdatingLayoutControl = React.memo((props: LayoutPinPropertyControlProps) => {
+  const { property, currentValue, updateFrame } = props
+  const pointInfo = useInspectorLayoutInfo(props.property)
+
+  const [mountCount, forceUpdate] = React.useReducer((c) => c + 1, 0)
```

Comment:
What's the deal with this? Can you add a comment explaining why we need this please

Reply:
I added a comment, and better yet, also fixed the variable name so it actually is self-explanatory